api/types/plugin.go

```diff
@@ -111,6 +113,21 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if (bearer.Token == "") == (bearer.TokenFile == "") {
 			return trace.BadParameter("exactly one of Token and TokenFile must be specified")
 		}
+	case *PluginSpecV1_Opsgenie:
+		if settings.Opsgenie == nil {
+			return trace.BadParameter("settings must be set")
+		}
+		if err := settings.Opsgenie.CheckAndSetDefaults(); err != nil {
+			return trace.Wrap(err)
+		}
+
+		bearer := p.Credentials.GetBearerToken()
+		if bearer == nil {
+			return trace.BadParameter("opsgenie plugin must be used with the bearer token credential type")
+		}
+		if (bearer.Token == "") == (bearer.TokenFile == "") {
```

Comment:
Both this and the Okta plugin add the same validation rule. Might be smart to factor this out to some `func (*PluginBearerTokenCredentials) CheckAndSetDefaults() error`

https://github.com/gravitational/teleport/pull/26349/files#diff-c05dcfe52aba6413a7c9443e99c3204098909bc87148471197e941693cb09614R132-R134
@mdwn